sandbox/cgroup/monitor.go

```diff
@@ -0,0 +1,151 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package cgroup
+
+import (
+	"errors"
+	"os"
+	"path"
+
+	"github.com/snapcore/snapd/sandbox/cgroup/inotify"
+)
+
+// appMonitorData contains all the data to monitor a specific Snap:
+// its name, the list of paths to monitor, and the channel to send
+// the notification when all the paths have been deleted.
+type appMonitorData struct {
+	name        string
+	cgroupPaths []string
+	npaths      int
```

Comment:
maybe this two should be a cgroupPaths map[string]bool and we remove the entries when the we get the deletion notifications?